nrf_security/src/CMakeLists.txt

```diff
@@ -279,6 +280,13 @@ if (CONFIG_MBEDTLS_TLS_LIBRARY)
     PUBLIC
       mbedcrypto_common
   )
+
+  target_include_directories(mbedcrypto_common
+      INTERFACE
+        ${ZEPHYR_BASE}/modules/mbedtls/include
+  )
+  zephyr_library_sources_ifdef(CONFIG_MBEDTLS_DEBUG ${ZEPHYR_BASE}/modules/mbedtls/debug.c)
+  zephyr_library_sources_ifdef(CONFIG_MBEDTLS_DEBUG ${ZEPHYR_BASE}/modules/mbedtls/zephyr_init.c)
```

Comment:
zephyr_init does more than call mbedtls_debug_set_threshold. It can also call mbedtls_memory_buffer_alloc_init for some configurations.

I assume we have our own mbedtls dynamic memory management somehow.

Are we certain a user cannot accidentally configure his build to also invoke zephyr_init's mbedtls_memory_buffer_alloc_init in a way that is incompatible with our own system?